### PR TITLE
Split protocol parameters into updatable and non-updateable

### LINKED ISSUE #264: Remove TODO's from chain and ledger spec
- Fix what can be easily fixed.
- Make issues and remove TODO's from the code for those issues that would take longer.

### LINKED ISSUE #343: Split protocol parameters into updatable and non-updateable
We have a set of parameters which can be updated through the standard update mechanism. Everything, in theory, can be updated through a hard fork, but we would also like to delineate a set of parameters which we expect to be updated during a hard fork.

We can keep these as exactly the same as the parameters, but with some indication that they cannot be updated through a regular update, and are tied to the major protocol version.

https://input-output-rnd.slack.com/archives/C4WQQKUU9/p1549272492276700?thread_ts=1548867374.270300&cid=C4WQQKUU9 has the history

After discussion, the result is to go back to having global parameters rather than non-updatable parameters. But we should have some explanations linking these to the major protocol version.

### PR DESCRIPTION
**Pull request by dnadales**
- Remove epoch change logic from update interface, as this is now handled by the chain spec.
- Reorganize the chain spec so that the related transition system definitions are closer to each other.
- Make the Byron chain and ledger spec match.
- Remove reference to `elens` and make `sEpoch` be an abstract function, to match what we want in the implementation: have the mapping from epoch to epoch lengths be a constant parameter to our validation functions.
- Remove labels from initial rules.
- Add explanations on how to map `upAdptThd` and `cfmThd` to `cardano-sl` protocol paramters.
- Update the voting rule to that the confirmation threshold is passed as an environment variable, and the `UPIVOTE` rule now multiplies `cfmThd` by the number of genesis keys.
- Remove questions as they are no longer relevant.
- Remove TODO as we seem to be OK with the constraint that on an update proposal the proposed maximum transaction size must be lower than the maximum block size.

Closes #343 
Closes #264

### CODE REVIEW

File: byron/chain/formal-spec/blockchain-spec.tex [hunk 1]
```diff
@@ -1124,18 +1157,30 @@ \section{Blockchain Extension}
...
-        \var{sgs} & \seqof{\VKeyGen} & \text{Last signers}\\
-        \var{h} & \Hash & \text{Current block hash} \\
+        \var{s_{last}} & \Slot & \text{slot of the last seen block} \\
+        \var{sgs} & \seqof{\VKeyGen} & \text{last signers}\\
+        \var{h} & \Hash & \text{current block hash} \\
         \var{utxo} & \UTxO & \text{UTxO} \\
-        \var{us} & \UPIState & \text{Update interface state} \\
-        \var{ds} & \DIState & \text{Delegation state}
+        \var{ds} & \DIState & \text{delegation state}\\
+        \var{us} & \UPIState & \text{update interface state} \\
```
Comment (ruhatch) [inline byron/chain/formal-spec/blockchain-spec.tex]: My favourite change is switching the order of these two ;)

File: byron/chain/formal-spec/blockchain-spec.tex [hunk 2]
```diff
@@ -548,18 +564,6 @@ \subsection{Counting signed blocks}
...
-  \emph{Protocol parameters}
-  \begin{equation*}
-    \begin{array}{r@{~\partialf~}l!{~\in~\ProtParams~}r}
-      \pp{stableAfter} & \mathbb{N} & \text{Chain stability parameter} \\
-      \pp{blockSignatureCountThreshold} & \left[\frac{1}{5}, \frac{1}{4}\right] & \text{Block signature count threshold} \\
-    \end{array}
-  \end{equation*}
-  \caption{Blockchain signature count protocol parameters}
-  \label{fig:defs:sigcnt}
-\end{figure}
```
Comment (nc6) [inline byron/chain/formal-spec/blockchain-spec.tex]: I'm not entirely seeing the reasoning behind this change. The aim was to split between updateable parameters (in the `pps` map) and non-updateable ones (which can be defined as abstract functions in much the same way as the other abstract functions). This change seems to do neither; instead, it just pushes the definition of these parameters up to the top level, where their context is much less understood.

I would rather either leave them as they are, or just add them to the "abstract functions" in this section, which would have the benefit of not changing much elsewhere.

File: byron/ledger/formal-spec/delegation.tex [hunk 3]
```diff
@@ -81,7 +81,7 @@ \section{Delegation}
         \mathcal{K} & \powerset{\VKeyGen} & \text{allowed delegators}\\
         \var{e} & \Epoch & \text{epoch}\\
         \var{s} & \Slot & \text{slot}\\
-        \var{k} & \SlotCount & \text{chain stability parameter}
+        \var{k} & \mathbb{N} & \text{chain stability parameter}
```
Comment (nc6) [inline byron/ledger/formal-spec/delegation.tex]: Why do we change this here? Surely this is a slot count?

File: byron/ledger/formal-spec/blockchain-interface.tex [hunk 4]
```diff
@@ -17,9 +17,10 @@ \subsection{Delegation interface}
     \left(
       \begin{array}{r@{~\in~}lr}
         \mathcal{K} & \powerset{\VKeyGen} & \text{allowed delegators}\\
-        \var{e} & \Epoch & \text{epoch}\\
-        \var{s} & \Slot & \text{slot}\\
-        \var{pps} & \PPMMap & \text{Protocol parameters}
+        k & \mathbb{N} & \text{chain stability parameter}\\
```
Comment (nc6) [inline byron/ledger/formal-spec/blockchain-interface.tex]: Again, I feel much less keen on these "env" parameters. What's the advantage of this?

File: byron/chain/formal-spec/blockchain-spec.tex (hunk 2 above)
Comment (nc6) [reply byron/chain/formal-spec/blockchain-spec.tex]: @ruhatch You may have opinions here?

**Comment by dnadales** [comment]
@nc6 I incorporated your changes, let me know what do you think about them now.

File: byron/chain/formal-spec/blockchain-spec.tex (hunk 1 above)
Comment (dnadales) [reply byron/chain/formal-spec/blockchain-spec.tex]: Yeah, it made my eyelid twitch :P 

Not only we bring consistency, but everybody knows that delegation comes before the update :D

**Comment by nc6** [comment]
LGTM!

**Pull request merged**